src/backend/joanie/tests/payment/test_backend_base.py

```diff
@@ -560,30 +595,42 @@ def test_payment_backend_base_do_on_refund(self):
         # Create payment and register it
         payment = {
             "id": "pay_0",
-            "amount": order.total,
+            "amount": Decimal(str(order.payment_schedule[0]["amount"])),
             "billing_address": billing_address,
             "installment_id": "d9356dd7-19a6-4695-b18e-ad93af41424a",
         }
 
         backend.call_do_on_payment_success(order, payment)
         Transaction.objects.get(reference="pay_0")
 
-        # - Order has been completed
-        self.assertEqual(order.state, enums.ORDER_STATE_COMPLETED)
-
-        # - Refund entirely the order
+        # - Order should be in state `pending_payment` since 1 or 2 installment has been paid
+        self.assertEqual(order.state, enums.ORDER_STATE_PENDING_PAYMENT)
+        order.flow.cancel()
+        self.assertEqual(order.state, enums.ORDER_STATE_CANCELED)
+        # - Refund the paid installment of the order in the payment schedule
         backend.call_do_on_refund(
-            amount=order.total,
+            amount=Decimal(str(order.payment_schedule[0]["amount"])),
             invoice=order.main_invoice,
             refund_reference="ref_0",
+            installment_id=payment["installment_id"],
+            is_transaction_canceled=False,
         )
-
-        # - Credit transaction has been created
+        # - Credit transaction has been created and a credit note
         self.assertEqual(
-            Transaction.objects.filter(reference="ref_0", total=-order.total).count(),
+            Transaction.objects.filter(
+                reference="ref_0",
+                total=-Decimal(str(order.payment_schedule[0]["amount"])),
+            ).count(),
             1,
         )
 
+        transaction = Transaction.objects.get(
+            reference="ref_0",
+            total=-Decimal(str(order.payment_schedule[0]["amount"])),
+        )
+
+        self.assertEqual(transaction.invoice.type, "credit_note")
+
         # - Order has been canceled
         order.refresh_from_db()
         self.assertEqual(order.state, "canceled")
```

Comment:
I don't understand why at the end we are in `canceled` state and not in `refund`. We have to discuss about that

Reply:
I've changed the behavior, now when all paid installments are refunded, the order's state goes from "refunding" to "refunded".